ports/activemq-cpp/portfile.cmake

```
@@ -8,92 +8,133 @@ vcpkg_download_distfile(ARCHIVE
     SHA512 83692d3dfd5ecf557fc88d204a03bf169ce6180bcff27be41b09409b8f7793368ffbeed42d98ef6374c6b6b477d9beb8a4a9ac584df9e56725ec59ceceaa6ae2
 )
 
-vcpkg_extract_source_archive_ex(
-    OUT_SOURCE_PATH SOURCE_PATH
-    ARCHIVE ${ARCHIVE}
-    PATCHES
-        FunctionLevelLinkingOn.diff
-        fix-crt-linkage.patch
-)
-
-set(ACTIVEMQCPP_MSVC_PROJ ${SOURCE_PATH}/vs2010-build/activemq-cpp.vcxproj)
+if (VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    vcpkg_extract_source_archive_ex(
+        OUT_SOURCE_PATH SOURCE_PATH
+        ARCHIVE ${ARCHIVE}
+    )
 
-string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" ACTIVEMQCPP_SHARED_LIB)
+    vcpkg_configure_make(
+        SOURCE_PATH ${SOURCE_PATH}
+        AUTOCONFIG
+        OPTIONS
+            "--with-openssl=${CURRENT_INSTALLED_DIR}"
+                "--with-apr=${CURRENT_INSTALLED_DIR}/tools/apr"
+    )
 
-if (ACTIVEMQCPP_SHARED_LIB)
-    set(RELEASE_CONF "ReleaseDLL")
-    set(DEBUG_CONF   "DebugDLL")
-    
-    set(ACTIVEMQCPP_LIB_PREFFIX )
-    set(ACTIVEMQCPP_LIB_SUFFIX d)
-    vcpkg_replace_string(${ACTIVEMQCPP_MSVC_PROJ} ";apr-1.lib" ";libapr-1.lib")
-else()
-    set(RELEASE_CONF "Release")
-    set(DEBUG_CONF   "Debug")
-    
-    set(ACTIVEMQCPP_LIB_PREFFIX lib)
-    set(ACTIVEMQCPP_LIB_SUFFIX )
-    vcpkg_replace_string(${ACTIVEMQCPP_MSVC_PROJ} ";libapr-1.lib" ";apr-1.lib")
-endif()
+    vcpkg_install_make()
+    file(RENAME ${CURRENT_PACKAGES_DIR}/include/activemq-cpp-${VERSION}/activemq ${CURRENT_PACKAGES_DIR}/include/activemq)
+    file(RENAME ${CURRENT_PACKAGES_DIR}/include/activemq-cpp-${VERSION}/cms ${CURRENT_PACKAGES_DIR}/include/cms)
+    file(RENAME ${CURRENT_PACKAGES_DIR}/include/activemq-cpp-${VERSION}/decaf ${CURRENT_PACKAGES_DIR}/include/decaf)
+    file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/include/activemq-cpp-${VERSION})
 
-if (VCPKG_TARGET_ARCHITECTURE MATCHES "x86")
-    set(BUILD_ARCH "Win32")
-    set(OUTPUT_DIR "Win32")
-elseif (VCPKG_TARGET_ARCHITECTURE MATCHES "x64")
-    set(BUILD_ARCH "x64")
-    set(OUTPUT_DIR "Win64")
+    vcpkg_copy_pdbs()
 else()
-    message(FATAL_ERROR "Unsupported architecture: ${VCPKG_TARGET_ARCHITECTURE}")
-endif()
+    if(VCPKG_CRT_LINKAGE STREQUAL "static")
+        vcpkg_extract_source_archive_ex(
+            OUT_SOURCE_PATH SOURCE_PATH
+            ARCHIVE ${ARCHIVE}
+            PATCHES
+                FunctionLevelLinkingOn.diff
+                fix-crt-linkage.patch
+        )
+    else()
+        vcpkg_extract_source_archive_ex(
+            OUT_SOURCE_PATH SOURCE_PATH
+            ARCHIVE ${ARCHIVE}
+            PATCHES
+                FunctionLevelLinkingOn.diff
+                fix-crt-linkage-dyn.patch
+        )
+    endif()

```

How about something like this?

```suggestion
    set(PATCHES FunctionLevelLinkingOn.diff)
    if(VCPKG_CRT_LINKAGE STREQUAL "static")
        list(APPEND PATCHES fix-crt-linkage.patch)
    else()
        list(APPEND PATCHES fix-crt-linkage-dyn.patch)
    endif()
    vcpkg_extract_source_archive_ex(
        OUT_SOURCE_PATH SOURCE_PATH
        ARCHIVE ${ARCHIVE}
        PATCHES
            ${PATCHES)
    )

```